duct.go

```diff
@@ -138,6 +156,12 @@ func (c *Composer) GetNetworkID() string {
 	return c.netID
 }
 
+type nullWriter struct{}
```

Comment:
wouldn't `ioutil.Discard` suitably work here? do we need our own type?

Reply:
oh i was looking for that, happy to replace thanks